### PR TITLE
Add docs and generated reference for Alpha configuration

### PR DESCRIPTION
**Pull request by JoelSpeed**
## Description

This PR introduces a code generator that reads the types in the options package and generates reference documentation for the types into the docs site.

## Motivation and Context

We are planning to introduce alpha structured config in this release, once that is done, we will want to document the schema for that. We can use this to generate a reference for the alpha config types when they have been introduced.

## How Has This Been Tested?

Generate code, test docs locally, seems to render nicely.

## Checklist:

- [x] My change requires a change to the documentation or CHANGELOG.
- [x] I have updated the documentation/CHANGELOG accordingly.
- [x] I have created a feature (non-master) branch for my PR.

### CODE REVIEW

**Comment by JoelSpeed** [comment]
One consideration we should probably discuss is whether this generator tool should become it's own repository 🤔

**Comment by JoelSpeed** [comment]
@NickMeves I've updated the documentation page for the Alpha config to explain that this is alpha and add warnings. Plus a brief bit about using it with conversion and the removed options. As we add new stuff to the alpha config we need to keep this up to date. I think this is a good start though. LMK what you think

File: go.mod
Comment (NickMeves) [inline go.mod:38]: Just wanted to confirm this was intended to get commited

Comment (JoelSpeed) [reply go.mod:38]: At present, the generator is sitting on a branch, so until that is merged to master, we need a replacement... How do you feel about merging this without merging the other to master?  🤔 Let me get a PR open

**Comment by JoelSpeed** [comment]
@NickMeves I updated the go.mod to point to the merged version of the reference generator tooling, this should be good to go assuming you're happy with the wording of the document

**Review by NickMeves — APPROVED** [review]

**Pull request merged**
